front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -227,70 +265,90 @@ const SpaceTimeChartWrapper = ({
-                if (!path) return zone;
-                const pathStyle = getPathStyle(
-                  hoveredItem,
-                  path,
-                  !!draggingState,
-                  subCategories,
-                  timetableItemsWithDetails,
-                  selectedTrainId
-                );
-                return {
-                  ...zone,
-                  color: pathStyle.color,
-                  size: pathStyle.level === 1 ? 2 : undefined,
-                };
-              })}
-              selectedTrainId={selectedTrainId}
-              onClose={() => onCloseOccupancyLayer?.(waypointId)}
-              topPadding={BASE_WAYPOINT_HEIGHT}
-            />
-          ),
-          manchetteNode: (
-            <TrackOccupancyManchette tracks={tracks || []}>
-              <div className="waypoint-wrapper flex justify-start">
-                <WaypointComponent
-                  waypoint={{
-                    id: waypointId,
-                    name: (
-                      <div className="d-flex flex-row align-items-center">
-                        {operationalPointName || operationalPointId}
-                        {loading && (
-                          <Spinner className="ml-2 small" spinnerClassName="spinner-border-sm" />
-                        )}
-                      </div>
-                    ),
-                    position: operationalPointPosition,
-                    onClick: handleWaypointClick,
-                  }}
-                  waypointRef={activeWaypointRef}
-                  isActive={false}
-                  isMenuActive={false}
-                />
-              </div>
-            </TrackOccupancyManchette>
-          ),
-        })
-      ) || []
-    );
-  }, [
-    trackOccupancyDiagramsData,
-    activeWaypointId,
-    timetableItemsWithDetails,
-    selectedTrainId,
-    hoveredItem,
-  ]);
+        }) => {
+          const baseZones = zones ?? [];
+          const zonesCountByPacedTrainId = countZonesByPacedTrainId(baseZones);
+
+          const styledZones = baseZones.map((zone) => {
+            const path = pathsById[zone.trainId];
+
+            const isHovered = hoveredPathId === zone.trainId;
+            const isSelected = selectedTrainId === zone.trainId;
+
+            const paced = toPaced(zone.trainId);
+            const totalOnTrack =
+              paced && zone.trackId
+                ? (zonesCountByPacedTrainId.get(paced)?.get(zone.trackId) ?? 0)
+                : 0;
+            if (!path) return zone;
+
+            const pathStyle = getPathStyle(
+              hoveredItem,
+              path,
+              !!draggingState,
+              subCategories,
+              timetableItemsWithDetails,
+              selectedTrainId
+            );
+            return {
+              ...zone,
+              label: isHovered ? `+${String(Math.max(0, totalOnTrack - 1))}` : undefined,
+              color: pathStyle?.color ?? zone.color,
```

Comment:
`pathStyle.color` is always defined ;)
```suggestion
              color: pathStyle.color,
```